src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/reporting/VisitCountsByDateReportService.kt

```diff
@@ -0,0 +1,170 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.service.reporting
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.PrisonDto
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.enums.VisitRestriction
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.enums.VisitStatus
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.reporting.SessionVisitCountsDto
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.sessions.SessionScheduleDto
+import uk.gov.justice.digital.hmpps.visitscheduler.model.entity.projections.VisitRestrictionStats
+import uk.gov.justice.digital.hmpps.visitscheduler.repository.SessionSlotRepository
+import uk.gov.justice.digital.hmpps.visitscheduler.repository.VisitRepository
+import uk.gov.justice.digital.hmpps.visitscheduler.service.PrisonsService
+import uk.gov.justice.digital.hmpps.visitscheduler.service.SessionService
+import uk.gov.justice.digital.hmpps.visitscheduler.service.TelemetryClientService
+import uk.gov.justice.digital.hmpps.visitscheduler.task.ReportingTask.Companion.LOG
+import java.time.LocalDate
+
+@Service
+class VisitCountsByDateReportService(
+  private val prisonsService: PrisonsService,
+  private val sessionService: SessionService,
+  private val telemetryClientService: TelemetryClientService,
+  private val visitRepository: VisitRepository,
+  private val sessionSlotRepository: SessionSlotRepository,
```

Comment:
Instead of accessing the repo directly, is there a SessionSlotService?

Reply:
Updated